verification/curator-service/ui/src/components/LinelistTable.tsx

```diff
@@ -41,12 +36,46 @@ interface Case {
     notes: string;
 }
 
-export default class LinelistTable extends React.Component {
+interface LinelistTableState {
+    tableRef: any,
+    url: string,
+}
+
+export default class LinelistTable extends React.Component<{}, LinelistTableState> {
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            tableRef: React.createRef(),
+            url: (process.env.REACT_APP_DATA_API_ENDPOINT || "") + '/api/cases/',
+        }
+    }
+
+     deleteCase(rowData: Case) {
+        return new Promise((reject) => {
+            let deleteUrl = this.state.url + rowData._id;
+            const response = axios.delete(deleteUrl);
+            response.then(() => {
+                // Refresh the table data
+                this.state.tableRef.current.onQueryChange();
+            }).catch((e) => {
+                reject(e);
+            });
+        })
+     }
 
+     editCase(newRowData: Case, oldRowData: Case | undefined) {
```

Comment:
In which occasion can we edit something undefined?

Reply:
TBH no idea, but it wouldn't build without that type declaration...

Reply:
ok thanks... let's keep that in mind, we should fail gracefully when we understand when that happens.